analysis/import/SonarImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/sonar/dataaccess/SonarMeasuresAPIDatasource.kt

```diff
@@ -51,44 +51,52 @@ class SonarMeasuresAPIDatasource(private val user: String, private val baseUrl:
     private val client: Client = ClientBuilder.newClient()
     private val logger = KotlinLogging.logger {}
 
+    private var measureBatches = 0
+    private var processedPages = 0
```

Comment:
I need some clarification here:
What are measureBatches and what are processPages? Why do we even have pages and why is there a page size?

Reply:
the sonar importer works by downloading information form a sonar instance. There not all information can be obtained at once, that's why they have pages. The same goes for the metrics; we will not obtain all of them at once, but we will obtain them in batches

Reply:
Do the pages include the metrics?

What type is the page size of 500? Is it bytes?

Are multiple batches adding up to one page? And multiple pages to the whole project?

Reply:
The page size is the number of files; each page contains a batch of metrics (subset of all metrics) for this files.